doc/src/start_states.md

```diff
@@ -0,0 +1,118 @@
+The following explains the syntax and semantics of Start States in lrlex.<br>
+A working example can be found in the repository at [lrpar/examples/start_states](https://github.com/softdevteam/grmtools/tree/master/lrpar/examples/start_states)
+
+# Motivation
+
+Start states are a feature from lex which can be used to add a bit of context sensitivity to lexical analysis.
+For instance, they can be used to implement nested comments (see the example in the repository).  Such that
+the tokens start/end markers of tokens maintain balance.
+
+This is achieved by making rules which are qualified to match only when the lexer is in a particular state.
+Additionally the lexer has a stack of states, and matching rules perform actions which modify the stack.
+
+# The INITIAL start state 
+By default unless specified otherwise all lex rules are members of the *INITIAL* start state.
```

Comment:
Can drop "By default"?

Reply:
Should be fixed in [d182e4a](https://github.com/softdevteam/grmtools/pull/468/commits/d182e4a8897848d1d6935e26471a13635665c3e8)